### PR TITLE
webapp node in azure explorer 

### PR DESCRIPTION
**Pull request by Eskibear**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppConfiguration.java [hunk 1]
```diff
@@ -60,11 +62,10 @@
...
     private static final String MISSING_ARTIFACT = "A web archive (.war) artifact has not been configured.";
     private static final String INVALID_WAR_FILE = "The artifact name %s is invalid. "
-        + "An artifact name may contain only the ASCII letters 'a' through 'z' (case-insensitive), "
-        + "the digits '0' through '9', '.', '-' and '_'.";
+            + "An artifact name may contain only the ASCII letters 'a' through 'z' (case-insensitive), "
+            + "the digits '0' through '9', '-' and '_'.";
 
-    private static final String WAR_NAME_REGEX = "^[.A-Za-z0-9_-]+\\.war$";
-    private static final String WEB_APP_CONFIGURATION_NODE = "AzureWebAppConfig";
+    private static final String WAR_NAME_REGEX = "^[A-Za-z0-9_-]+\\.war$";
```
Comment (jdneo) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppConfiguration.java]: . is allowed here.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppConfiguration.java [hunk 2]
```diff
@@ -60,11 +62,10 @@
     private static final String MISSING_PRICING_TIER = "Pricing Tier not provided.";
     private static final String MISSING_ARTIFACT = "A web archive (.war) artifact has not been configured.";
     private static final String INVALID_WAR_FILE = "The artifact name %s is invalid. "
-        + "An artifact name may contain only the ASCII letters 'a' through 'z' (case-insensitive), "
-        + "the digits '0' through '9', '.', '-' and '_'.";
+            + "An artifact name may contain only the ASCII letters 'a' through 'z' (case-insensitive), "
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppConfiguration.java]: also include "."

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java [hunk 3]
```diff
@@ -326,6 +439,17 @@ public void cleanWebAppsOnLinux() {
         subscriptionIdToWebAppsOnLinuxMap.clear();
     }
 
+    private Azure getAzureBySid(String sid) throws IOException {
```
Comment (jdneo) [inline Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java]: better to move to AuthMethodManager and become public

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/WebAppModule.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    protected void refreshItems() throws AzureCmdException {
+        webAppModulePresenter.onModuleRefresh();
+
+    }
+
+    @Override
+    public void removeNode(String sid, String id, Node node) {
+        try {
+            webAppModulePresenter.onDeleteWebApp(sid, id);
+            removeDirectChildNode(node);
```
Comment (adashen) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/WebAppModule.java]: should removeDirectChildNode called by presenter?

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/WebAppModulePresenter.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+    public static void onStopWebApp(String sid, String id) throws IOException {
+        AzureWebAppMvpModel.getInstance().stopWebApp(sid, id);
+    }
+
+    /**
+     * Called from view when the view needs refresh.
+     */
+    public void onModuleRefresh() {
+        List<ResourceEx<WebApp>> winApps = AzureWebAppMvpModel.getInstance().listWebApps(true);
```
Comment (adashen) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/WebAppModulePresenter.java]: blocked call?

**Pull request merged**